src/Functions/abs.cpp

```diff
@@ -45,7 +45,7 @@ template <> struct FunctionUnaryArithmeticMonotonicity<NameAbs>
         if ((left_float < 0 && right_float > 0) || (left_float > 0 && right_float < 0))
             return {};
 
-        return { .is_monotonic = true, .is_positive = left_float > 0, .is_strict = true, };
+        return { .is_monotonic = true, .is_positive = std::min(left_float, right_float) >= 0, .is_strict = true, };
```

Comment:
If both boundaries are greater or equals zero it is positive; if both less or equal zero it is negative; otherwise it is not monotonic.

Reply:
I think the implementation already works as you describe?

The previous condition makes sure that both boundaries are either >=0 or <= 0:

```cpp
if ((left_float < 0 && right_float > 0) || (left_float > 0 && right_float < 0))
            return {};
```

Then `.is_positive = std::min(left_float, right_float) >= 0` means it's positive if the lower boundary >= 0.

Reply:
That's true, thank you!